src/wiktextract/extractor/de/gloss.py

```diff
@@ -3,58 +3,167 @@
+            gloss_text = extract_tags_from_gloss_text(gloss_data, gloss_text)
 
-            page_data[-1]["senses"].append(gloss_data)
+            if gloss_text or not sub_glosses_list_nodes:
+                gloss_data["glosses"] = [gloss_text]
+                page_data[-1]["senses"].append(gloss_data)
+
+            for sub_list_node in sub_glosses_list_nodes:
+                process_gloss_list_item(
+                    wxr,
+                    page_data,
+                    sub_list_node,
+                    senseid,
+                    gloss_data if not gloss_text else None,
+                )
 
         else:
             wxr.wtp.debug(
                 f"Unexpected list item in glosses: {list_item_node}",
                 sortid="extractor/de/glosses/extract_glosses/29",
             )
             continue
+
+
+def handle_sense_modifier(wxr, list_item_node):
+    wxr.wtp.debug(
+        f"Skipped a sense modifier in gloss list: {list_item_node}",
+        sortid="extractor/de/glosses/extract_glosses/19",
+    )
+    # XXX: We should extract the modifier. However, it seems to affect
+    # multiple glosses. Needs investigation.
+    pass
+
+
+def process_K_template(
+    wxr: WiktextractContext,
+    gloss_data: defaultdict(list),
+    list_item_node: NodeKind.LIST_ITEM,
+) -> None:
+    for template_node in list_item_node.find_child(NodeKind.TEMPLATE):
+        if template_node.template_name == "K":
+            categories = []
+
+            temp_node = copy.deepcopy(template_node)
+            for key, value in template_node.template_parameters.items():
+                if isinstance(key, int):
+                    temp_node.largs = temp_node.largs[:1] + [[value]]
+                    # Cleaned K template will always end with ":". Remove it.
+                    category = clean_node(wxr, {}, temp_node)[:-1]
+                    if category:
+                        categories.append(category)
+                if key == "ft":
+                    # ft (free text) is used liberally to modify the context
+                    # template. Sometimes it seems to belong rather to the
+                    # gloss itself. Most of the time it is not useful.
+                    # XXX Treat free text in K templates.
+                    continue
+
+            prep = template_node.template_parameters.get("Prä")
+            case = template_node.template_parameters.get("Kas")
+            category = (prep if prep else "") + (" + " + case if case else "")
+            if category:
+                categories.append(category)
+
+            # Remove the template_node from the children of list_item_node
+            list_item_node.children = [
+                c for c in list_item_node.children if c != template_node
+            ]
+
+            gloss_data["tags"].extend(categories)
+
+
+def extract_tags_from_gloss_text(
+    gloss_data: defaultdict(list), gloss_text: str
+) -> None:
+    parts = gloss_text.split(":", 1)
```

Comment:
I've added some examples from real wikitext.

And it seems you're right. That might be a better strategy. However, I would rather move on at the moment since what I have works fairly well. I added an XXX comment to not forget about having another look at this.